studio/components/grid/components/footer/pagination/Pagination.tsx

```diff
@@ -32,26 +32,63 @@ const Pagination: React.FC<PaginationProps> = () => {
   const [page, setPage] = React.useState(state.page);
   const maxPages = Math.ceil(state.totalRows / state.rowsPerPage);
   const totalPages = state.totalRows > 0 ? maxPages : 1;
+  const { selectedRows } = state;
 
   React.useEffect(() => {
     if (state.page != page) setPage(state.page);
   }, [state.page, page]);
 
   function onPreviousPage() {
-    if (state.page > 1) {
+    if (state.page <= maxPages) {
       const previousPage = state.page - 1;
       setPage(previousPage);
       dispatch({ type: 'SET_PAGE', payload: previousPage });
     }
   }
 
+  if (selectedRows.size >= 1) {
```

Comment:
having this conditional at the root of component is gonna have the logic run on render - is this intended?